sys-fs/bcachefs-tools/bcachefs-tools-24_p20221120.ebuild

```
@@ -0,0 +1,259 @@
+HOMEPAGE="https://bcachefs.org/"
+if [[ "${PV}" = "9999" ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://evilpiepirate.org/git/bcachefs-tools.git"
+	SRC_URI="$(cargo_crate_uris ${CRATES})"
+else
+	MY_PV=f1f88825c371f84edb85a156de5e1962503d23b2
+	SRC_URI="https://evilpiepirate.org/git/bcachefs-tools.git/snapshot/bcachefs-tools-${MY_PV}.tar.zst
+		$(cargo_crate_uris ${CRATES})"
+	S="${WORKDIR}/${PN}-${MY_PV}"
+	KEYWORDS="~amd64"
+fi
+
+SLOT="0"
+LICENSE="Apache-2.0 BSD GPL-2 MIT"
+IUSE="fuse test"
+RESTRICT="!test? ( test )"
+
+DEPEND="
+	app-arch/lz4
+	app-arch/zstd
+	dev-libs/libaio
+	dev-libs/libsodium
+	dev-libs/userspace-rcu
+	sys-apps/keyutils
+	sys-apps/util-linux
+	sys-libs/zlib
+	virtual/udev
+	fuse? ( >=sys-fs/fuse-3.7.0 )
+"
+
+BDEPEND="
+	${PYTHON_DEPS}
+	$(python_gen_any_dep '
+		dev-python/docutils[${PYTHON_USEDEP}]
+		test? (
+			dev-python/pytest[${PYTHON_USEDEP}]
+			dev-python/pytest-xdist[${PYTHON_USEDEP}]
+		)
+	')
+	app-arch/zstd
+	sys-devel/clang
+	virtual/rust
+"
+
+python_check_deps() {
+	if use test; then
+		python_has_version \
+			"dev-python/pytest[${PYTHON_USEDEP}]" \
+			"dev-python/pytest-xdist[${PYTHON_USEDEP}]"
+	fi
+	python_has_version "dev-python/docutils[${PYTHON_USEDEP}]"
+
+}
+
+pkg_setup() {
+	python-any-r1_pkg_setup
+}
+
+src_unpack() {
+	unpacker bcachefs-tools-${MY_PV}.tar.zst
+	cargo_src_unpack
+}
+
+src_prepare() {
+	default
+	tc-export CC
+	sed \
+		-e '/^CFLAGS/s:-O2::' \
+		-e '/^CFLAGS/s:-g::' \
+		-e 's:pytest-3:/bin/true:g' \
+		-i Makefile || die
+	append-lfs-flags
+}
+
+src_compile() {
+	if use fuse; then
+		export BCACHEFS_FUSE=1
+	fi
+	export VERSION=${PV}
+	default
+	# Rust UUID-based mounter isn't in 'all' target, may as well use ebuild functions
+	for module in bch_bindgen mount; do
+		pushd "${S}/rust-src/$module"
+			LIBBCACHEFS_LIB="${S}" LIBBCACHEFS_INCLUDE="${S}" cargo_src_compile
+		popd
+	done
+	ln -f "${S}/rust-src/mount/target/release/bcachefs-mount" "${S}/mount.bcachefs" || die
+	if use test; then
+		emake tests
+	fi
+}
+
+src_test() {
+	if ! use fuse; then
+		EPYTEST_IGNORE=( tests/test_fuse.py )
+	fi
+	EPYTEST_DESELECT=(
+		# Valgrind
+		'tests/test_fixture.py::test_read_after_free'
+		'tests/test_fixture.py::test_undefined'
+		'tests/test_fixture.py::test_write_after_free'
+		'tests/test_fixture.py::test_undefined_branch'
+		'tests/test_fixture.py::test_leak'
+		'tests/test_fixture.py::test_check'
+		# Fails in portage
+		'tests/test_basic.py::test_format'
+		'tests/test_basic.py::test_fsck'
+		'tests/test_basic.py::test_list'
+		'tests/test_basic.py::test_list_inodes'
+		'tests/test_basic.py::test_list_dirent'
+	)
+	epytest -v -n "$(makeopts_jobs)"
+}
+
+src_install() {
+	QA_FLAGS_IGNORED="/usr/bin/mount.bcachefs"
+	# Raised upstream; we don't expect anything to link against this outside of bcachefs-tools bins, for now
+	QA_SONAME=".*libbcachefs.so"
+	exeinto /usr/bin
+	for file in bcachefs fsck.bcachefs mkfs.bcachefs mount.bcachefs mount.bcachefs.sh; do
```

Always `local`ize vars so they don't leak:
```suggestion
	local
	for file in bcachefs fsck.bcachefs mkfs.bcachefs mount.bcachefs mount.bcachefs.sh; do
```